### PR TITLE
Update github.com/openfga/go-sdk to 0.6.3

### PR DESCRIPTION
**Pull request by yakupt-canonical**
## Description

This PR updates the OpenFGA SDK version to 0.6.3 to enable OpenTelemetry support.
Also ability to provide custom HTTP client is added for propagation of OpenTelemetry trace IDs between systems.

## Engineering checklist
_Check only items that apply_

- [ ] Documentation updated
- [x] Covered by unit tests
- [ ] Covered by integration tests
- [ ] Independent change*

 _* This is an independent change if it can be merged and released indepently without any related service deployments._

## Merging instructions

The preferred way of merging:
- [ ] Merge
- [x] Squash and Merge
- [ ] Rebase and Merge

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go-version: [ '1.21' ]
+        go-version: [ '1.22' ]
```
Comment (yakupt-canonical) [inline .github/workflows/ci.yml:12]: This upgrade needed for gosec checks.

File: client.go [hunk 2]
```diff
@@ -94,6 +101,26 @@ func NewClient(ctx context.Context, p OpenFGAParams) (*Client, error) {
 				ApiToken: p.Token,
 			},
 		}
+	} else {
+		config.Credentials = &credentials.Credentials{
+			Method: credentials.CredentialsMethodNone,
+		}
+	}
+	if p.HTTPClient != nil {
```
Comment (andrei-polukhin) [inline client.go:109]: Curiously, should we have the `else` check for ensuring that if a client is not provided, the default one is used?

**Review by andrei-polukhin — APPROVED** [review]
Looks good, thanks!

File: client.go [hunk 3]
```diff
@@ -94,6 +101,26 @@ func NewClient(ctx context.Context, p OpenFGAParams) (*Client, error) {
...
+		defaultHeaders := make(map[string]string)
+		if len(headers) != 0 {
+			for idx := range headers {
+				defaultHeaders[headers[idx].Key] = headers[idx].Value
+			}
+		}
+		config.DefaultHeaders = defaultHeaders
+	}
+	if p.Telemetry != nil {
+		config.Telemetry = p.Telemetry
```
Comment (brotich) [inline client.go:123]: consider adding an example with the telemetry configured.

**Review by brotich — APPROVED** [review]

File: client.go (hunk 3 above)
Comment (yakupt-canonical) [reply client.go:123]: Added, thank you.

File: example_test.go [hunk 4]
```diff
@@ -6,6 +6,7 @@ package ofga_test
 import (
 	"context"
 	"fmt"
+	"github.com/openfga/go-sdk/telemetry"
 	"os"
 	"testing"
```
[the remark this answers is not in the document]
Comment (yakupt-canonical) [reply example_test.go]: I forgot to do that, thank you.

**Pull request merged**